scripts/spell-check.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/sh
+#
+## Copyright (C) 2020 The Squid Software Foundation and contributors
+##
+## Squid software is distributed under GPLv2+ license and includes
+## contributions from numerous individuals and organizations.
+## Please see the COPYING and CONTRIBUTORS files for details.
+##
+
+#
+# This script runs codespell against selected files.
+#
+# Use -q or --quiet option to run quietly
+#
+
+CODESPELL_LOC=`which codespell`
+if test "${CODESPELL_LOC}" = ""; then
+    echo "This script requires codespell which was not found."
+    exit 1
+fi
+
+CODESPELL_VER=`codespell --version 2>&1`
+echo "The codespell version is ${CODESPELL_VER}."
+
+UNSTAGED_CHANGES=`git diff | wc -l`
+if test "${UNSTAGED_CHANGES}" != "0"; then
+    echo "There are unstaged changes. Stage these first to prevent conflict."
+    exit 1
+fi	
+
+WHITE_LIST=scripts/codespell-whitelist.txt
+if test ! -f "${WHITE_LIST}"; then
```

Comment:
What happens if this test is removed and `codespell-whitelist.txt` is missing? If codespell quits with an error, then perhaps we can remove this check? Please note that you may need to surround `${WHITE_LIST}` in quotes where you call codespell so that codespell gets an empty string as a parameter value.

Reply:
Without this test, codespell gives the following message and the dumps a long usage page before exiting the script.
"ERROR: cannot find ignore-words file: scripts/xxcodespell-whitelist.txt"
The current result seems cleaner. 
I will wait for a response before removing the test.
And yes, the quotes are needed in case of an empty string (the script goes wild without them).

Reply:
> ... dumps a long usage page before exiting the script.

I agree that your current solution is appropriate due to this noisy usage dump. I consider this change request resolved. Thank you.